pkg/verifiedcar/verifiedcar.go

```diff
@@ -136,68 +135,108 @@ func (cfg Config) VerifyBlockStream(ctx context.Context, cbr BlockReader, lsys l
 		return 0, 0, traversalError(err)
 	}
 
+	if nbro.Error != nil {
+		// capture any errors not bubbled up through the traversal, i.e. see
+		// https://github.com/ipld/go-ipld-prime/pull/524
+		return 0, 0, nbro.Error
+	}
+
 	// make sure we don't have any extraneous data beyond what the traversal needs
 	_, err = cbr.Next()
-	if !errors.Is(err, io.EOF) {
+	if ctx.Err() == nil && !errors.Is(err, context.Canceled) && !errors.Is(err, io.EOF) {
```

Comment:
don't we also need to return a seperate error for context cancelled? it's not a success no?